database/src/base/tx_indexer.rs

```diff
@@ -6,6 +6,12 @@ pub trait TxIndexerDbManager {
         block_height: u64,
     ) -> anyhow::Result<()>;
 
+    async fn check_transaction_save_correctly(
```

Comment:
My suggestion to rename this function into `validate_saved_transaction_deserializable` to be more explicit of what exactly we want to check. Add comments for future us about the problem we solve by having this.